### PR TITLE
enable prettier

### PR DESCRIPTION
**Pull request by chenxsan**
Supersede https://github.com/webpack/webpack.js.org/pull/3109 as there's no news so far from @montogeek https://github.com/webpack/webpack.js.org/pull/3109#issuecomment-752839686, thus I'll take care of it.

I've made some changes to `.markdownlint.json` as prettier doesn't provide options for markdown formatting.

### CODE REVIEW

**Comment by alexander-akait** [comment]
It is awesome

File: src/components/Configuration/components.js [hunk 1]
```diff
@@ -28,17 +28,17 @@ const Card = ({ body }) => {
...
 Card.propTypes = {
-  body: PropTypes.node
+  body: PropTypes.node,
 };
 export class Details extends Component {
   static propTypes = {
     url: PropTypes.string,
-    children: PropTypes.arrayOf(PropTypes.node)
-  }
+    myChilds: PropTypes.arrayOf(PropTypes.node),
```
Comment (montogeek) [inline src/components/Configuration/components.js:36]: Why this change?

File: src/content/api/cli.md [hunk 2]
```diff
@@ -293,21 +293,21 @@ You can also pass multiple values:
...
 
-You can merge two or more different webpack configurations with the help of `--merge`: 
+You can merge two or more different webpack configurations with the help of `--merge`:
 
 ```bash
 npx webpack --config ./first.js --config ./second.js --merge
 ```
 
-__Print result of webpack as a JSON__
+**Print result of webpack as a JSON**
```
Comment (montogeek) [inline src/content/api/cli.md:304]: This was done by prettier?

Comment (montogeek) [reply src/content/api/cli.md:304]: It doesnt change anything visually.

File: src/components/Configuration/components.js (hunk 1 above)
Comment (chenxsan) [reply src/components/Configuration/components.js:36]: image

File: src/content/api/cli.md (hunk 2 above)
Comment (chenxsan) [reply src/content/api/cli.md:304]: It is. And there's no option to force it to use `__`, that's why I've made changes to `.markdownlint.json`.

File: src/components/Configuration/components.js (hunk 1 above)
Comment (montogeek) [reply src/components/Configuration/components.js:36]: Is this a new warning?
Or you want to fix it in this PR?

Comment (chenxsan) [reply src/components/Configuration/components.js:36]: Not a new one, before there's a `// eslint-disable-next-line` to disable the warning, see https://github.com/webpack/webpack.js.org/blob/master/src/components/Configuration/Configuration.jsx#L11

File: prettier.config.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  singleQuote: true,
+  semi: true,
```
Comment (alexander-akait) [inline prettier.config.js:3]: You can omit this, it is `true` by default

File: prettier.config.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  singleQuote: true,
```
Comment (alexander-akait) [inline prettier.config.js:2]: Why? You can use JSX, so using `"` is preferred here

File: prettier.config.js (hunk 3 above)
Comment (chenxsan) [reply prettier.config.js:3]: Indeed. But let's keep it as it's explicit and can serve as a guide for anyone who want to check. I believe not everyone knows what're the default values of prettier's options.

File: prettier.config.js (hunk 4 above)
Comment (chenxsan) [reply prettier.config.js:2]: I have no preference here. I think `singleQuote` was used more often in code before, hence I set it `true`.

Comment (chenxsan) [reply prettier.config.js]: Updated.

Comment (chenxsan) [reply prettier.config.js:2]: Unfortunately there's too many errors hard to fix with `singleQuote: false`, thus I'm reverting the change.

Comment (alexander-akait) [reply prettier.config.js:2]: We can return to this latest, let's integrate prettier firstly
